### PR TITLE
Add pyhive

### PR DESCRIPTION
**Pull request by AbdealiJK**
This is a revival of https://github.com/conda-forge/staged-recipes/pull/841

Now sasl has been added to conda-forge at: https://github.com/conda-forge/sasl-feedstock and https://github.com/conda-forge/thrift_sasl-feedstock

### CODE REVIEW

**Comment by xhochy** [comment]
The build fails because `sasl` is not available on Windows thus you need to make this for now a macOS- and Linux-only recipe.

**Comment by xhochy** [comment]
Alternatively you might try the solution of  `thrift_sasl` here too: https://github.com/conda-forge/thrift_sasl-feedstock/blob/master/recipe/meta.yaml#L28-L29

File: recipes/pyhive/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - requests >=1.0.0
+
+test:
+  imports:
+    - "{{ name }}"
+
+about:
+  home: https://github.com/dropbox/PyHive
+  license: Apache 2.0
+  summary: Collection of Python DB-API and SQLAlchemy interfaces for Presto and Hive.
```
Comment (CJ-Wright) [inline recipes/pyhive/meta.yaml]: need a licence, and a licence file

File: recipes/pyhive/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  run:
+    - future
+    - sasl >=0.2.1
+    - thrift >=0.10.0
+    - thrift_sasl >=0.1.0
+    - requests >=1.0.0
+
+test:
+  imports:
+    - "{{ name }}"
```
Comment (CJ-Wright) [inline recipes/pyhive/meta.yaml]: please hard code this test and add a few more if possible.

**Pull request closed without merging**